app/models/concerns/host_common.rb

```diff
@@ -224,9 +224,15 @@ def cnt_hosts(config_group)
   end
 
   def update_config_group_counters(record)
-    record.update_attribute(:hostgroups_count, cnt_hostgroups(record))
-    record.update_attribute(:hosts_count, cnt_hosts(record))
+    record.update_attribute(:hostgroups_count, cnt_hostgroups(record) + creation_delta_for(:hostgroup))
+    record.update_attribute(:hosts_count, cnt_hosts(record) + creation_delta_for(:host))
 
     record.update_puppetclasses_total_hosts
   end
+
+  def creation_delta_for(host_or_hostgroup)
+    #on remove this would still work because new_record is false.
+    klass = {:host => Host::Base, :hostgroup => Hostgroup}[host_or_hostgroup]
```

Comment:
What's the point in this?  Can't you just pass Host::Base/Hostgroup into the method?